docs/upgrading-to-integration.asciidoc

```diff
@@ -0,0 +1,167 @@
+[[upgrade-to-apm-integration]]
+=== Upgrade to the Elastic APM integration
+
+* <<why-upgrade-to-integrations>>
+* <<apm-arch-upgrade>>
+* <<apm-integration-upgrade-limitations>>
+* <<apm-integration-upgrade-steps>>
+* <<apm-integration-upgrade-steps-ess>>
+
+[discrete]
+[[why-upgrade-to-integrations]]
+=== Why upgrade?
+
+The new APM ecosystem offers a number of benefits:
+
+**{fleet}**:
+
+* A single, unified way to add monitoring for logs, metrics, traces, and other types of data to each host -- install one thing instead of multiple
+* Central, unified configuration management -- no need to edit multiple configuration files
+
+**Data streams**:
+
+* Reduced number of fields per index, better space efficiency, and faster queries
+* More granular data control
+* Errors and metrics data streams are shared with other data sources -- which means better long-term integration with the logs and metrics apps
+* Removes template inheritance for ILM policies and makes use of new {es} index and component templates
+* Fixes <<server-resource-exists-not-alias>>
+
+**APM Integration**:
+
+* Easier to install APM on edge machines
+* Improved source map handling and APM agent configuration management
+* Less configuration
+* Easier and less error-prone upgrade path
```

Comment:
do we want to add zero-downtime configuration changes when apm-server is managed? @simitt

Reply:
yeah that's a good point